### PR TITLE
Reader: Remove follows that are under the wrong keys

### PR DESCRIPTION
**Pull request by blowery**
Not quite sure how this happens, but jan's seen it. This should clean it up when sync completes.

### CODE REVIEW

File: client/state/reader/follows/reducer.js [hunk 1]
```diff
@@ -99,7 +99,11 @@ export const items = createReducer(
 			let urlKey = prepareComparableUrl( action.payload.feedUrl );
 			const newValues = { is_following: true };
 
-			const actualFeedUrl = get( action.payload, [ 'follow', 'feed_URL' ], action.payload.feedUrl );
+			const actualFeedUrl = get(
```
Comment (blowery) [inline client/state/reader/follows/reducer.js]: just pretty

File: client/state/reader/follows/reducer.js [hunk 2]
```diff
@@ -177,7 +181,12 @@ export const items = createReducer(
 			// Only check items with an ID (the subscription ID) because those are what
 			// we show on the manage listing. Items without an ID are either inflight follows
 			// or follows that we picked up from a feed, site, or post object.
-			return omitBy( state, follow => follow.ID && ! seenSubscriptions.has( follow.feed_URL ) );
+			return omitBy(
```
Comment (blowery) [inline client/state/reader/follows/reducer.js]: actual diffs!

File: client/state/reader/follows/test/reducer.js [hunk 3]
```diff
@@ -70,7 +70,9 @@ describe( 'reducer', () => {
 				type: READER_RECORD_UNFOLLOW,
 				payload: { url: 'http://discover.wordpress.com' },
 			} );
-			expect( state[ 'discover.wordpress.com' ] ).to.eql( { blog_ID: 123, is_following: false } );
+			expect( state[ 'discover.wordpress.com' ] ).to.eql(
```
Comment (blowery) [inline client/state/reader/follows/test/reducer.js]: so pretty

File: client/state/reader/follows/test/reducer.js [hunk 4]
```diff
@@ -483,7 +485,10 @@ describe( 'reducer', () => {
 			} );
 			const state = items( original, {
 				type: READER_FOLLOW_ERROR,
-				payload: { feedUrl: 'http://discoverinvalid.wordpress.com', error: 'invalid_feed' },
+				payload: {
```
Comment (blowery) [inline client/state/reader/follows/test/reducer.js]: pretty? pretty.

File: client/state/reader/follows/test/reducer.js [hunk 5]
```diff
@@ -644,5 +649,41 @@ describe( 'reducer', () => {
 				},
 			} );
 		} );
+
+		it( 'should remove sites under the wrong key', () => {
```
Comment (blowery) [inline client/state/reader/follows/test/reducer.js]: this is new! and also pretty.

File: client/state/reader/follows/reducer.js [hunk 6]
```diff
@@ -177,7 +181,12 @@ export const items = createReducer(
...
+				state,
+				( follow, key ) =>
+					key !== prepareComparableUrl( follow.feed_URL ) ||
```
Comment (samouri) [inline client/state/reader/follows/reducer.js]: how were we getting into this state?

Comment (blowery) [inline client/state/reader/follows/reducer.js]: i have no idea.

Comment (blowery) [inline client/state/reader/follows/reducer.js]: maybe an earlier screw up in how we unfollowed things? or how we moved from one feed url to a new one?

**Comment by blowery** [comment]
closing for now as we fixed the underlying bug earlier

**Comment by samouri** [comment]
i think a valuable "belt and suspenders" approach would be to force a cache clear every N days (5?).

**Pull request closed without merging**